### ISSUE TITLE
Smallest successful corpus size?

### ISSUE THREAD

**Issue by dannybtran** [ibab/tensorflow-wavenet#212]
What's the most success anyone has had with smaller sized corpora? I've tried with 90sec of extracted voice audio, but getting mostly static even after 20k-100k steps (loss of ~1-2)

**Comment by dannybtran** [comment]
Let me rephrase, the wav form has a speech like rhythm, but it's so distorted, it sounds like static.

**Comment by akademi4eg** [comment]
https://soundcloud.com/dmytro-tkanov/wavenet-10sec-train?in=dmytro-tkanov/sets/wavenet-results

This sound is generated after training until convergence to 0.5 loss on 10 sec of audio.

**Comment by dannybtran** [comment]
@akademi4eg very interesting. So assuming 16khz and 16-bit that's about a 320k wav file?  Did you generate that with fast-gen turned on?  I found that if I disable fast generation, that quality improves.

**Comment by akademi4eg** [comment]
Unfortunately, I don't remember what was the size of wav-file and don't have access to it anymore.
Fast generation was turned on. As far as I understand, fast generation should produce exactly same sound as "slow" version if random seed is same. Idea of fast generation is to store and reuse results of convolutions for previous samples.

**Comment by lemonzi** [comment]
You should be able to train with any corpus size, as long as you train long enough and the model is big enough. The issues should come with larger datasets, not smaller. If you train with a small dataset you will get overfitting, but it should still work. Actually, the unit tests generate a sine wave and train on it until the network perfectly recreates them. It could be that the model that you get is very sensitive to the initial seed...

Using fast generation should indeed be equivalent (and there is, or used to be, a unit test that checks for that), it just changes the order of operations to be able to cache some results.

**Comment by dannybtran** [comment]
Thanks for the info.  Closing this now.
